core/templates/dev/head/pages/exploration_player/ExplorationPlayerStateService.ts

```diff
@@ -147,11 +181,22 @@ oppia.factory('ExplorationPlayerStateService', [
           initExplorationPlayer(callback);
         }
       },
+      initializeQuestionPlayer: function(config, callback) {
+        PlayerTranscriptService.init();
+        initQuestionPlayer(config, callback);
+      },
       getCurrentEngineService: function() {
         return currentEngineService;
       },
       isInPretestMode: function() {
-        return inPretestMode;
+        return explorationMode === EXPLORATION_MODE.PRETEST;
+      },
+      isInQuestionMode: function() {
+        return explorationMode === EXPLORATION_MODE.PRETEST ||
+        explorationMode === EXPLORATION_MODE.QUESTION_PLAYER;
+      },
+      isInQuestionPlayerMode: function() {
```

Comment:
We have a similar method in ContextServices, right? (isInQuestionContext)

Reply:
Yes that's true, please see the above conversation with akshay as to why we have two mode methods (Also note, we already use this for pretest vs exploration, I just basically extended that for the question player as well)